src/CTTConfig.py

```diff
@@ -0,0 +1,163 @@
+
+        job.add_argument('--rootfs', help='Path to your rootfs image')
+        job.add_argument('--kernel', help='Path to your kernel image')
+        job.add_argument('--dtb', help='Path to your dtb')
+        job.add_argument('--dtb-folder', help='Path to your dtb folder')
+        job.add_argument('--modules', help='Path to your modules tar.gz')
+        job.add_argument('-t', '--tests', nargs='+',
+                         help='Tests to run on the board')
+        job.add_argument('-m', '--tests-multinode', nargs='+',
+                         help='Multinode tests to run')
+
+        lava = parser.add_argument_group('LAVA server options')
+        lava.add_argument('--server', help='LAVA server to send results to')
+        lava.add_argument('--username', help='LAVA username')
+        lava.add_argument('--token', help='LAVA token')
+
+        artifacts = parser.add_argument_group('Artifacts options')
+        artifacts.add_argument('--api-token', help='KernelCI API token')
+        artifacts.add_argument('--tree', default='mainline',
+                               help='KernelCI tree to use')
+        artifacts.add_argument('--branch', default='master',
+                               help='KernelCI branch to use')
+        artifacts.add_argument('--defconfigs', nargs='+',
+                               help='KernelCI defconfig to use')
+        artifacts.add_argument('--no-kci', action='store_true',
+                               help='Ignore KernelCI artifacts')
+
+        ssh = parser.add_argument_group('SSH server options')
+        ssh.add_argument('--ssh-server', help='SSH server')
+        ssh.add_argument('--ssh-username', help='SSH username')
+
+        self.__cmdline = vars(parser.parse_args())
+
+    def __validate_config_file(self):
+        if not self.__config.has_section(DEFAULT_SECTION):
+            raise SectionError('Missing %s section' % DEFAULT_SECTION)
+
+    def __validate_cmdline(self):
+        # We can always just list the boards
+        if self.__cmdline['list']:
+            return
+
+        # Validate that we have all the basic options...
+        for option in mandatory_keys:
+            if not self.__contains__(option):
+                raise OptionError('Missing %s option' % option)
+
+        # ... and the options that can only be passed through the command line
+        if self.__cmdline['boards'] is None:
+            raise OptionError('Missing board')
+
+        if (self.__cmdline['tests_multinode'] is None and
+                self.__cmdline['tests'] is None):
+            raise OptionError('Missing test')
+
+        for board in self.__cmdline['boards']:
+            # Expand all boards to what we know
+            if board == 'all':
+                self.__cmdline['boards'] = [b for b in boards.boards.keys()]
+                break
+
+            if board not in boards.boards.keys():
+                raise OptionError('Invalid board')
```

Comment:
Since this will stop the command, it could be a great idea to display which board is the invalid one. Otherwise, it will be awful to find it if we provided more than two or three boards.